src/block/Hopper.php

```diff
@@ -57,6 +57,17 @@ public function getStateBitmask() : int{
 		return 0b1111;
 	}
 
+	public function getFacing() : int{ return $this->facing; }
+
+	/** @return $this */
+	public function setFacing(int $facing) : self{
+		if($facing === Facing::UP){
+			throw new \InvalidArgumentException("Hopper may not face upward");
+		}
+		$this->facing = $facing;
+		return $this;
```

Comment:
Is there a particular reason to make this a fluent setter?

Reply:
It's standard.